src/Microsoft.ML.Data/Evaluators/EvaluatorUtils.cs

```diff
@@ -615,7 +615,7 @@ public static void ReconcileKeyValues(IHostEnvironment env, IDataView[] views, s
             var keyNamesVBuffer = new VBuffer<ReadOnlyMemory<char>>(keyNames.Count, keyNames.Keys.ToArray());
             ValueGetter<VBuffer<ReadOnlyMemory<char>>> keyValueGetter =
                     (ref VBuffer<ReadOnlyMemory<char>> dst) =>
-                        dst = new VBuffer<ReadOnlyMemory<char>>(keyNamesVBuffer.Length, keyNamesVBuffer.Count, keyNamesVBuffer.Values, keyNamesVBuffer.Indices);
+                        keyNamesVBuffer.CopyTo(ref dst);
```

Comment:
>keyNamesVBuffer [](start = 24, length = 15)

Wow. This was a really bad bug. We shared the underlying output buffers among everyone that could have asked for this!!

The more I read of your PR the more enthusiastic I become. I had thought only of the effort it would save me and others going forward of having to catch all these potential misuses of `VBuffer`, but now I become alarmed at the sheer volume of stuff I seem to have missed. :P